clean-modular-code/activity-3/clean-code-activity-3.md

```diff
@@ -15,48 +15,50 @@ kernelspec:
 
+Things like helpful error messages and fast failing will also improve the experience for someone else using your code - OR your future self. 
+ 
 :::{tip}
 Using functions, classes, and methods (functions within a class) is a great first step in handling messy data. A function or method provides a modular unit you can test outside of the workflow for the edge cases you may encounter. Also, because a function is a modular unit, you can add elements to handle unexpected processing features as you build your workflow.
-:::
 
-something about debuggers? 
-* https://jupyterlab.readthedocs.io/en/stable/user/debugger.html
+Once you have these functions and methods, you can add checks using conditional statements and try/except blocks that anticipate edge cases and errors that you may encounter when processing your data. 
+:::
 
 ## Manage the unexpected 
 
-In this activity, you will apply the following strategies:
-
-* [conditional statements](../checks-conditionals/python-conditionals)
-* try/except blocks
-
-to process the JOSS citation data. 
+In this activity, you will apply the following strategies to make your code more robust, maintainable & usable:
 
-:::{todo}
-What branch is the lesson with try/except // ask for forgiveness, checks elements in??
-IN THIS PR:
-https://github.com/pyOpenSci/lessons/pull/14/files#diff-7f4ff1b75e85d38f3955cca051e68e8746773c279b34c9a0a400b9c2dc1240ff 
-:::
+* **[Fail fast with useful error messages](fail-fast)**: Failing fast is a software engineering term that means allowing your 
+  code to stop early when something goes wrong, ensuring that errors are caught 
+  and communicated clearly. This helps the user quickly understand what went 
+  wrong and where.
+* Use **[conditional statements](../checks-conditionals/python-conditionals)** 
+  to check for specific conditions before executing certain parts. This allows you to create specific workflows that your code followed based on specific conditions.
+* **[Try/except blocks](../checks-conditionals/python-function-checks)** allow 
+  you to handle potential errors by attempting an operation and catching any 
+  exceptions if they occur, providing useful feedback. In some cases you may want the program to end on an error. In other cases, you may want to handle it in a specific way. 
 
-When you can, try to use the Pythonic approach of asking for forgiveness later (ie use try/except blocks) rather than conditional statements.
+When you can, try to use a [Pythonic approach](pythonic-checks) to catch errors early; this means asking for forgiveness later (often using try/except blocks) vs. using conditional statements to check an object's state or type.
 
 ```{code-cell} ipython3
 ---
 editable: true
 slideshow:
   slide_type: ''
 ---
-# This works but is less pythonic
+# This works but is less Pythonic as it's a look before you leap approach
```

Comment:
```suggestion
# This works but is less Pythonic as it's a "look before you leap" approach since it does an extra check before returning the title
```